Added Firebase and prepared Vuex store

jeffbuonamici opened the pull request:
Attached firebase to the project. Also initiated state data with getter methods.

Started playing around with the UI

jeffbuonamici committed:
Attached Firebase

jeffbuonamici committed:
Prepared Vuex store for authorization

jeffbuonamici committed:
Updating the UI

jeffbuonamici committed:
Created basic search bar component

jeffbuonamici committed:
Fixed App.vue bug

MikeSmvl commented:
The firebase and Vuex store code is good, but the offset change on the signup.vue and signin.vue is making them less centered. I changed the layout to a "justify-center" which fixed our issues.

Commit:
Update Signin.vue

Signin.vue had some layout issues when opened in Safari. The grid system for the elements were changed to have it correctly centered in all browsers.

Commit:
Update Signup.vue

Signup.vue had some layout issues when opened in Safari. The grid system for the elements were changed to have it correctly centered in all browsers.
